src/components/BlogPost.tsx

```diff
@@ -71,7 +50,9 @@ export const BlogPost: React.FC<BlogPostProps> = ({
             </div>
           </div>
           {
-            ((isAuthed() && isAuthor()) || (isAuthed() && isAdmin())) && <BlogPostActionsMenu id={id} title={title} content={content} isAdmin={isAdmin()}/>
+            ((currUser.data !== null) && (currUser.data !== undefined)) &&
+            ((isAuthed(status) && isAuthor(currUser.data, author)) || (isAuthed(status) && isAdmin(currUser.data))) &&
```

Comment:
Is it possible to pull this into one helper function? The logic is getting a bit confusing and long for an in-line check